### PR TITLE
hw/util: blinker/beeper implementation

### PR DESCRIPTION
**Pull request by sdalu**
Provide an implementation of a blinker/beeper, only need to provide an on/off method (gpio, pwm, ...).
Make it easy to provide a simple user feedback

### CODE REVIEW

File: hw/util/blinker/include/blinker/blink.h [hunk 1]
```diff
@@ -0,0 +1,285 @@
+#ifndef _BLINK_H_
```
Comment (mkiiskila) [inline hw/util/blinker/include/blinker/blink.h]: Need apache license blurb.

File: hw/util/blinker/src/blink.c [hunk 2]
```diff
@@ -0,0 +1,350 @@
+#include <assert.h>
```
Comment (mkiiskila) [inline hw/util/blinker/src/blink.c]: Also needs the apache license comment.

File: hw/util/blinker/src/blink.c [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    if (blink->next.scheduled) {
+        if ((count == 0                  ) ||     /* Continuous blink */
+            (count == blink->running.step) ||     /* End of group     */
+            (blink->next.scheduled == BLINK_SCHEDULED_WAIT_BLINK)) {
+            return false;
+        }
+    }
+    
+    /* Off state */
+    if        (count == 0) {                            /* Continuous blink */
```
Comment (mkiiskila) [inline hw/util/blinker/src/blink.c]: Coding style: there seems to be stray whitespaces here after 'if'.

File: hw/util/blinker/src/blink.c [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    return true;
+}
+
+/*
+ * Mark the blink handle as stopped (ie: not running).
+ *
+ * @param blink                 blink handler
+ */
+static inline void
+blink_mark_stopped(blink_t *blink, bool forced) {
```
Comment (mkiiskila) [inline hw/util/blinker/src/blink.c]: Coding style: { on next line on it's own after after function declaration.

File: hw/util/blinker/src/blink.c [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+{
+    uint16_t b_on, b_off;
+    
+    /* Check for next blink */
+    if (blink->running.onoff(blink, &b_on, &b_off)) {
+        blink->running.flags &= ~(BLINK_FLG_SCHEDULE_ON |
+                                  BLINK_FLG_SCHEDULE_OFF);
+
+        /* Infinite on */
+        if        (b_on  == BLINK_TIME_FOREVER) {
```
Comment (mkiiskila) [inline hw/util/blinker/src/blink.c]: Coding style: extraneous whitespaces here after if as well.

File: hw/util/blinker/include/blinker/blink.h [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+     */
+    struct os_mutex mutex;
+    /*
+     * Callout for 'on'
+     */
+    struct os_callout on_callout;
+    /*
+     * Callout for 'off'
+     */
+    struct os_callout off_callout;
```
Comment (mkiiskila) [inline hw/util/blinker/include/blinker/blink.h]: You could save some space by replacing one of these callouts with a state variable, and keeping state of the 'on/off' state that way.

**Comment by sdalu** [comment]
Coding style has been fixed, and now only using one callout

**Pull request merged**